ui/selectors/selectors.js

```diff
@@ -280,17 +283,38 @@ export const getMetaMaskAccounts = createSelector(
   getInternalAccounts,
   getMetaMaskAccountBalances,
   getMetaMaskCachedBalances,
-  (internalAccounts, balances, cachedBalances) =>
+  getMultichainBalances,
+  getMultichainNetwork,
+  (
+    internalAccounts,
+    balances,
+    cachedBalances,
+    multichainBalances,
+    multichainNetwork,
+  ) =>
     Object.values(internalAccounts).reduce((accounts, internalAccount) => {
       // TODO: mix in the identity state here as well, consolidating this
       // selector with `accountsWithSendEtherInfoSelector`
       let account = internalAccount;
 
-      if (balances[internalAccount.address]) {
+      // TODO: `AccountTracker` balances are in hex and `MultichainBalance` are in number.
+      // We should consolidate the format to either hex or number
+      if (
+        isEvmAccountType(internalAccount.type) ||
+        balances[internalAccount.address]
+      ) {
         account = {
           ...account,
           ...balances[internalAccount.address],
         };
+      } else {
+        account = {
+          ...account,
+          balance:
+            multichainBalances?.[internalAccount.id]?.[
+              MultichainNativeAssets[multichainNetwork.chainId]
+            ]?.amount ?? '0',
+        };
```

Comment:
I feel like this is not matching the original behavior, maybe we should do something like:

```suggestion
      if (isEvmAccountType(internalAccount.type)) {
        if (balances[internalAccount.address]) {
          account = {
            ...account,
            ...balances[internalAccount.address],
          };
        }
      } else {
        account = {
          ...account,
          balance:
            multichainBalances?.[internalAccount.id]?.[
              MultichainNativeAssets[multichainNetwork.chainId]
            ]?.amount ?? '0',
        };
```

This make sure that we we're gonna "set" the `account` (for EVM accounts) only when the balance is set.

Since `balances[internalAccount.address]` might effectively be `undefined` (while this will be accepted by the spread operator, this feels a bit fragile to me), WDYT?